src/main/scala/temple/generate/language/service/go/GoServiceGenerator.scala

```diff
@@ -96,22 +107,71 @@ object GoServiceGenerator extends ServiceGenerator {
     sb.toString
   }
 
+  private def generateJsonMiddleware(): String =
+    FileUtils.readFile("src/main/scala/temple/generate/language/service/go/genFiles/JsonMiddleware.go")
+
+  private def generateHandler(serviceName: String, endpoint: Endpoint): String =
+    s"func ${serviceName + endpoint}Handler(w http.ResponseWriter, r *http.Request) {}\n\n"
+
+  private def generateHandlers(serviceName: String, endpoints: Set[Endpoint]): String = {
+    val sb = new StringBuilder
+    if (endpoints.contains(Endpoint.ReadAll)) {
+      sb.append(generateHandler(serviceName, Endpoint.ReadAll))
+    }
+    if (endpoints.contains(Endpoint.Create)) {
+      sb.append(generateHandler(serviceName, Endpoint.Create))
+    }
+    if (endpoints.contains(Endpoint.Read)) {
+      sb.append(generateHandler(serviceName, Endpoint.Read))
+    }
+    if (endpoints.contains(Endpoint.Update)) {
+      sb.append(generateHandler(serviceName, Endpoint.Update))
+    }
+    if (endpoints.contains(Endpoint.Delete)) {
+      sb.append(generateHandler(serviceName, Endpoint.Delete))
+    }
```

Comment:
I see what you're doing here... I'm wondering if we could define a `sort` for these and just call `generateHandler` in a loop?

Reply:
One simple way, if `values` is defined as a list in Endpoint’s companion object
```scala
for (endpoint <- Endpoint.values if endpoints contains endpoint) {
  sb.append(generateHandler(serviceName, Endpoint.Delete))
}
```
where

Reply:
Does `.values` preserve an ordering?

Reply:
It would need to be defined manually, so just make it a `Seq`

Reply:
> One simple way, if `values` is defined as a list in Endpoint’s companion object
> 
> ```scala
> for (endpoint <- Endpoint.values if endpoints contains endpoint) {
>   sb.append(generateHandler(serviceName, Endpoint.Delete))
> }
> ```

Nice solution, didn't know about this filtering notation